odd-collector-sdk/odd_collector_sdk/utils/collector_config_generation.py

```diff
@@ -0,0 +1,121 @@
+from pathlib import Path
+from typing import Union
+
+from odd_collector_sdk.logger import logger
+from odd_collector_sdk.errors import LoadConfigError
+from odd_collector_sdk.types import PluginFactory
+from odd_collector_sdk.utils.yaml_parser import parse_yaml
+from odd_collector_sdk.domain.collector_config import CollectorConfig
+
+
+def read_config_yaml(collector_config_path: Union[str, Path]) -> dict:
+    """
+    Read and parse a collector configuration .yaml file.
+
+    Args:
+        collector_config_path: an absolute path to the collector configuration file.
+
+    Returns:
+        Parsed configuration as a dictionary.
+    """
+    try:
+        config_path = Path(collector_config_path).resolve()
+        logger.debug(f"{config_path=}")
+        logger.info("Start reading config")
+
+        with open(config_path) as f:
+            parsed = parse_yaml(f)
+
+        return parsed
+    except Exception as e:
+        raise LoadConfigError(e)
+
+
+def unpack_config_logical_sections(
+    parsed_config: dict,
+) -> tuple[dict, dict, list[dict]]:
+    """
+    Unpack information into a logical sections from a parsed configuration.
+
+    Parameters:
+        parsed_config: Parsed configuration as a dictionary.
+
+    Returns:
+        A tuple containing information about secrets backend and kwargs,
+        collector settings, and plugins.
+    """
+    plugins = parsed_config.pop("plugins", [])
+    secrets_info = {
+        "secrets_backend": parsed_config.pop("secrets_backend", None),
+        "secrets_backend_kwargs": parsed_config.pop("secrets_backend_kwargs", {}),
+    }
+    return secrets_info, parsed_config, plugins
+
+
+def generate_collector_config(
+    collector_settings: dict,
+    plugins: list[dict],
```

Comment:
Let's create a service dataclass for plugin. Right now it's difficult to understand what plugin is and I have to check config example each time to make sure if I treat dict correctly. Plugin comparing to secrets backend has strict set of settings and we can use it to parse and create the dataclass.

Reply:
I'm actually sure that we should have the plugin dataclass somewhere in sdk but I didn't check it

Reply:
We already have Plugin dataclass in odd_collector_sdk/domain/plugin.py for type hinting:
```
class Plugin(BaseSettings, extra=Extra.allow):
    type: str
    name: str
    description: Optional[str] = None
    namespace: Optional[str] = None
```
But I am not sure that it is good to use it here to unpack using it, because it is the "base" one. We are using PLUGIN_FACTORY from odd_collector/domain/plugin.py (or other odd_collector repo) and this factory contains mapper to all used plugins with more detailed structure and validation, like:
```
class PostgreSQLPlugin(DatabasePlugin):
    type: Literal["postgresql"]
    port: int = 5432
    database: str
    password: SecretStr = SecretStr("")
    schemas_filter: Filter = Filter()
```
So in the result of `plugin_factory[plugin["type"]].parse_obj(plugin) for plugin in plugins` we will get the list of different Plugin classes.

Reply:
I would stick to leave it as it is right now.